csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -909,6 +915,237 @@ private static StructArray GetTableSchemas(
                 nullBitmapBuffer.Build());
         }
 
+        internal async Task<TGetCatalogsResp> GetCatalogsAsync(CancellationToken cancellationToken)
+        {
+            if (SessionHandle == null)
+            {
+                throw new InvalidOperationException("Invalid session");
+            }
+
+            TGetCatalogsReq req = new TGetCatalogsReq(SessionHandle);
+            if (AreResultsAvailableDirectly())
+            {
+                SetDirectResults(req);
+            }
+
+            TGetCatalogsResp resp = await Client.GetCatalogs(req, cancellationToken);
+            if (resp.Status.StatusCode != TStatusCode.SUCCESS_STATUS)
+            {
+                throw new HiveServer2Exception(resp.Status.ErrorMessage)
+                    .SetNativeError(resp.Status.ErrorCode)
+                    .SetSqlState(resp.Status.SqlState);
+            }
+
+            return resp;
+        }
+
+        internal async Task<TGetSchemasResp> GetSchemasAsync(string? catalogName, string? schemaName, CancellationToken cancellationToken)
+        {
+            if (SessionHandle == null)
+            {
+                throw new InvalidOperationException("Invalid session");
+            }
+
+            TGetSchemasReq req = new(SessionHandle);
+            if (AreResultsAvailableDirectly())
+            {
+                SetDirectResults(req);
+            }
+            if (!string.IsNullOrEmpty(catalogName))
```

Comment:
Could there be a difference in semantics between "value not set", "value is null" and "value is blank" when it comes to setting these parameter values? For instance, maybe "value is blank" matches only a blank catalog name while "value not set" matches any catalog name? Obviously we can't distinguish between all three possibilities in C# without doing something extraordinary, but I wonder whether it's correct to treat null and blank identically.

Reply:
(Applies to all these parameters, of course, and not just catalog name.)